torchaudio/functional/functional.py

```diff
@@ -1373,9 +1370,12 @@ def _get_sinc_resample_kernel(
         # at specific positions, not over a regular grid.
         if resampling_method == "sinc_interpolation":
             window = torch.cos(t * math.pi / lowpass_filter_width / 2)**2
-        elif resampling_method == "kaiser_window":
-            beta = torch.tensor(beta, dtype=float)
-            window = torch.i0(beta * torch.sqrt(1 - (t / lowpass_filter_width) ** 2)) / torch.i0(beta)
+        else:
+            # kaiser_window
+            if beta is None:
+                beta = 14.769656459379492
+            beta_tensor = torch.tensor(float(beta))
```

Comment:
😮 Nicely done. I did not know optional type refinement can be done this elegantly.